include/internal/catch_default_main.hpp

```diff
@@ -13,12 +13,20 @@
 
 #ifndef __OBJC__
 
+#ifndef CATCH_INTERNAL_CDECL
+#ifdef _MSC_VER
```

Comment:
IMO it would be a good idea to additionally constrain this even more to just 32bit x86 with the [predefined macro _M_IX86](https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170).

> `_M_IX86` Defined as the integer literal value 600 for compilations that target x86 processors. This macro isn't defined for x64 or ARM compilation targets.

Reply:
I believe the convention, when specifying __cdecl, is always to specify it though rather than making it arch-specific. (I believe the requirement is for all architectures, it's just technically no different on some so it doesn't cause a problem in practice if it's not there.)

For example:
"On ARM and x64 processors, __cdecl is accepted but typically ignored by the compiler."
https://docs.microsoft.com/en-us/cpp/cpp/cdecl?view=msvc-170